contrib/pkg/broker/server/server.go

```diff
@@ -146,19 +164,30 @@ func (s *server) bind(w http.ResponseWriter, r *http.Request) {
 	if result, err := s.controller.Bind(instanceID, bindingID, &req); err == nil {
 		util.WriteResponse(w, http.StatusOK, result)
 	} else {
-		util.WriteResponse(w, http.StatusBadRequest, err)
+		util.WriteErrorResponse(w, http.StatusBadRequest, err)
 	}
 }
 
 func (s *server) unBind(w http.ResponseWriter, r *http.Request) {
 	instanceID := mux.Vars(r)["instance_id"]
 	bindingID := mux.Vars(r)["binding_id"]
+	q := r.URL.Query()
+	serviceID := first(q["service_id"])
+	planID := first(q["plan_id"])
 	glog.Infof("UnBind: Service instance guid: %s:%s", bindingID, instanceID)
 
-	if err := s.controller.UnBind(instanceID, bindingID); err == nil {
+	if err := s.controller.UnBind(instanceID, bindingID, serviceID, planID); err == nil {
 		w.WriteHeader(http.StatusOK)
 		fmt.Print(w, "{}") //id)
 	} else {
-		util.WriteResponse(w, http.StatusBadRequest, err)
+		util.WriteErrorResponse(w, http.StatusBadRequest, err)
+	}
+}
+
+func first(values []string) string {
```

Comment:
This should not be necessary. If you just use q.Get(whatever) it returns the first value or ""